### PR TITLE
refactor(dropdown-select): surf-1898

### PR DESCRIPTION
**Pull request by lalithkarikelli**
JIRA: SURF-1898

### LGTM's
* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: src/scss/components/dropdown-select/_config.scss [hunk 1]
```diff
@@ -4,6 +4,10 @@
 
 /// ========== MIXINS ========== ///
 @import "mixins";
+@import "./../form/mixins/fieldName";
```
Comment (CITguy) [inline src/scss/components/dropdown-select/_config.scss]: please remove `./` from beginning of import path (it is redundant)

File: src/scss/components/dropdown-select/SelectFacade.scss [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectControl.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  background-color: $gray-0;
+  color: $gray-900;
+
+  @if $focused {
+    &:-moz-reset-focusring {
+      background-color: $gray-900;
+    }
+
+    outline: none;
+  }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectControl.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+@mixin __selectControl--invalid($focused: false) {
+  background-color: $gray-0;
+  color: $gray-900;
+
+  @if $focused {
+    &:-moz-reset-focusring {
+      background-color: $gray-900;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectControl.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+@mixin __selectControl--disabled($focused: false) {
+  &:-moz-reset-focusring {
+    background-color: $gray-900;
+  }
+
+  outline: none;
+  @if not $focused {
+    background-color: $gray-100;
+    color: $gray-700;
+  }
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectControl.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+// .SelectControl(invalid)        -> hxSelectControl(invalid);
+// .SelectControl(invalid-focus)  -> hxSelectControl(invalid, $focused: true);
+// .SelectControl(disabled)       -> hxSelectControl(disabled);
+// .SelectControl(disabled-focus) -> hxSelectControl(disabled, $focused: true);
+@mixin hxSelectControl($state: null, $focused: false) {
+  @if $state == invalid {
+    @include __selectControl--invalid($focused);
+  } @else if $state == disabled {
+    @include __selectControl--disabled($focused);
+  } @else {
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectControl.scss [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+*/
+
+@mixin __selectControl($focused: false) {
+  @if $focused {
+    &:-moz-reset-focusring {
+      background-color: $gray-900;
+    }
+
+    outline: none;
+  } @else {
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectFacade.scss [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    border: 2px solid $red-900;
+    box-shadow: $focus-glow-invalid;
+  }
+}
+
+@mixin __selectFacade--disabled($focused: false) {
+  border: 1px solid $gray-500;
+  box-shadow: $focus-glow;
+  @if not $focused {
+    --hxTrigger-backgroundColor: $gray-100;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectFacade.scss [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  }
+  @else {
+    border: 1px solid $gray-500;
+    color: $gray-900;
+  }
+}
+
+@mixin __selectFacade--invalid($focused: false) {
+  border: 2px solid $red-900;
+  color: $gray-900;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/SelectFacade.scss [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+  @if $focused {
+    border: 2px solid $red-900;
+    box-shadow: $focus-glow-invalid;
+  }
+}
+
+@mixin __selectFacade--disabled($focused: false) {
+  border: 1px solid $gray-500;
+  box-shadow: $focus-glow;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/SelectFacade.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/_config.scss [hunk 11]
```diff
@@ -4,6 +4,10 @@
...
+@import "./SelectControl.scss";
+@import "./SelectFacade.scss";
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/_config.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss [hunk 12]
```diff
@@ -0,0 +1,143 @@
+@import "vars";
+
+// ===== MIXINS ==========
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss [hunk 13]
```diff
@@ -0,0 +1,143 @@
...
+  }
+
+  > select {
+    border: none;
+    padding-left: 0.75rem;
+    width: 100%; // prevent overflow for lengthy option text
+
+    // Adjust native trigger for MS browsers (mainly IE)
+    // Edge will obscure it with <hx-select>.
+
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss [hunk 14]
```diff
@@ -0,0 +1,143 @@
...
+    &:-ms-set-expand {
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/_config.scss (hunk 1 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/_config.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss (hunk 13 above)
[the remark this answers is not in the document]
Comment (CITguy) [reply src/scss/components/dropdown-select/index.scss]: I'm still seeing a blank line above `&::-ms-set-expand`

File: src/scss/components/dropdown-select/index.scss [hunk 15]
```diff
@@ -70,74 +66,74 @@ hx-select-control {
     border: none;
     padding-left: 0.75rem;
     width: 100%; // prevent overflow for lengthy option text
-
     // Adjust native trigger for MS browsers (mainly IE)
     // Edge will obscure it with <hx-select>.
-    #MS.set.expand({
+
+    &::-ms-set-expand {
```
Comment (CITguy) [inline src/scss/components/dropdown-select/index.scss]: Incorrect pseudo-element. Please change to `&::-ms-expand`

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 16]
```diff
@@ -0,0 +1,63 @@
+// Mixins to style a native <select> control.
+
+@mixin __selectControl($focused: false) {
+  @if $focused {
+    &:-moz-reset-focusring {
```
Comment (CITguy) [inline src/scss/components/dropdown-select/mixins/SelectControl.scss]: Incorrect pseudo element. Please change to `&:-moz-focusring`

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  @else {
+    background-color: $gray-0;
+    color: $gray-900;
+  }
+}
+
+@mixin __selectControl--disabled($focused: false) {
+  @if $focused {
+    &:-moz-reset-focusring {
```
Comment (CITguy) [inline src/scss/components/dropdown-select/mixins/SelectControl.scss]: Incorrect pseudo element. Please change to `&:-moz-focusring`

File: src/scss/components/dropdown-select/index.scss (hunk 13 above)
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss (hunk 15 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss (hunk 16 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss (hunk 17 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: docs/components/dropdown-select/_spec.scss [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+      @include hxSelectControl($focused: true);
+
+      ~ hx-select {
+        @include hxSelectFacade($focused: true);
+      }
+
+      /* ----- Disabled --------------- */
+      &:disabled {
+        @include hxSelectControl(disabled, $focused: true);
+        ~ hx-select {
```
Comment (CITguy) [inline docs/components/dropdown-select/_spec.scss]: Please add blank line between `@include` and `~ hx-select`...

```scss
&:disabled {
  @include ...

  ~ hx-select { 
    ...
  }
}
```

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+  @else {
+    background-color: $gray-0;
+    color: $gray-900;
+  }
+}
+
+@mixin __selectControl--disabled($focused: false) {
+  @if $focused {
+    &:-moz-focusring {
+      background-color: $gray-900;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 20]
```diff
@@ -0,0 +1,63 @@
+// Mixins to style a native <select> control.
+
+@mixin __selectControl($focused: false) {
+  @if $focused {
+    &:-moz-focusring {
+      background-color: $gray-900;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    outline: none;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 22]
```diff
@@ -0,0 +1,63 @@
...
+}
+
+@mixin __selectControl--invalid($focused: false) {
+  @if $focused {
+    &:-moz-focusring {
+      color: transparent;
+      text-shadow: 0 0 0 $gray-900;
+    }
+
+    outline: none;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/mixins/SelectControl.scss [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+  }
+}
+
+@mixin __selectControl--disabled($focused: false) {
+  @if $focused {
+    &:-moz-focusring {
+      background-color: $gray-900;
+    }
+
+    outline: none;
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/mixins/SelectControl.scss]: Changes have been done, please review.

File: docs/components/dropdown-select/_spec.scss [hunk 24]
```diff
@@ -0,0 +1,33 @@
+@import "components/dropdown-select/mixins/SelectControl";
+@import "components/dropdown-select/mixins/SelectFacade";
```
[the remark this answers is not in the document]
Comment (lalithkarikelli) [reply docs/components/dropdown-select/_spec.scss]: Changes have been done, please review.

File: docs/components/dropdown-select/_spec.scss (hunk 18 above)
Comment (lalithkarikelli) [reply docs/components/dropdown-select/_spec.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/_config.scss [hunk 25]
```diff
@@ -4,6 +4,10 @@
 
 /// ========== MIXINS ========== ///
 @import "mixins";
+@import "../form/mixins/fieldName";
+@import "./mixins/SelectControl.scss";
+@import "./mixins/SelectFacade.scss";
```
Comment (CITguy) [inline src/scss/components/dropdown-select/_config.scss]: Please remove `.scss` file extension. They're unnecessary.

File: src/scss/components/dropdown-select/index.scss [hunk 26]
```diff
@@ -1,9 +1,5 @@
-@import (reference) 'vars';
-
-// ===== MIXINS ==========
-@import (reference) './SelectControl';
-@import (reference) './SelectFacade';
-
+@import "vars";
```
Comment (CITguy) [inline src/scss/components/dropdown-select/index.scss]: Please remove "vars" import. It is already imported at the top of `_config.scss`.

File: src/scss/components/dropdown-select/index.scss [hunk 27]
```diff
@@ -70,74 +66,73 @@ hx-select-control {
...
 }
 
 // ----- PRISTINE ---------------
 hx-select-control > select {
-  .SelectControl(pristine);
+  @include hxSelectControl;
 
   ~ label {
-    .FieldName();
+    @include hxFieldName();
```
Comment (CITguy) [inline src/scss/components/dropdown-select/index.scss]: remove parens...

```scss
@include hxFieldName;
```

**Comment by CITguy** [comment]
Please rebase when you have the chance.

File: src/scss/components/dropdown-select/index.scss (hunk 26 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/_config.scss (hunk 25 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/_config.scss]: Changes have been done, please review.

File: src/scss/components/dropdown-select/index.scss (hunk 27 above)
Comment (lalithkarikelli) [reply src/scss/components/dropdown-select/index.scss]: Changes have been done, please review.

**Comment by CITguy** [comment]
Dev LGTM

**Comment by CITguy** [comment]
Please rebase.

**Comment by lalithkarikelli** [comment]
> Please rebase.

Rebase done ... Please merge the PR

File: docs/docs.less [hunk 28]
```diff
@@ -1,10 +1,10 @@
 /* ==================================================== *\
  * Styles below are NOT distributed with HelixUI assets *
 \* ==================================================== */
-@import (reference) 'vars';
-@import (reference) 'components/box/index';
+@import (reference) "vars";
```
Comment (CITguy) [inline docs/docs.less]: Please revert the quote changes back to single-quote. Let's keep the PR changes to a minimum.

File: src/less/components.less [hunk 29]
```diff
@@ -23,7 +23,6 @@
     - Complex CSS Selectors
     - NO !important
 \* ========== COMPONENTS ========== */
-@import 'components/button/index';
-@import 'components/checkbox/index';
-@import 'components/dropdown-select/index';
-@import 'components/form/index';
+@import "components/button/index";
+@import "components/checkbox/index";
+@import "components/form/index";
```
Comment (CITguy) [inline src/less/components.less]: Please revert the quote changes back to single-quote. Let's keep the PR changes to a minimum.

Comment (lalithkarikelli) [reply src/less/components.less]: Changes have been done, please review.

File: docs/docs.less (hunk 28 above)
Comment (lalithkarikelli) [reply docs/docs.less]: Changes have been done, please review.

File: src/scss/components/dropdown-select/_config.scss [hunk 30]
```diff
@@ -4,6 +4,10 @@
 
 /// ========== MIXINS ========== ///
 @import "mixins";
+@import "../form/mixins/fieldName";
+@import "./mixins/SelectControl";
+@import "./mixins/SelectFacade";
+
 
 // LESS               SCSS
```
Comment (100stacks) [inline src/scss/components/dropdown-select/_config.scss]: Please move this mixin into it's own file -> `components/dropdown-select/mixins/SelectReset.scss`.

File: src/scss/components/dropdown-select/index.scss [hunk 31]
```diff
@@ -70,74 +65,73 @@ hx-select-control {
...
+    @include hxSelectFacade(disabled);
   }
 
   &:focus {
-    .SelectControl(disabled-focus);
+    @include hxSelectControl(disabled, $focused: true);
 
     ~ hx-select {
-      .SelectFacade(disabled-focus);
+      @include hxSelectFacade(disabled, $focused: true);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/index.scss]: Please change to:

```scss
@include hxSelectFacade(disabled, true);
```

File: src/scss/components/dropdown-select/index.scss [hunk 32]
```diff
@@ -70,74 +65,73 @@ hx-select-control {
...
+    @include hxSelectFacade(invalid);
   }
 
   &:focus {
-    .SelectControl(invalid-focus);
+    @include hxSelectControl(invalid, $focused: true);
 
     ~ hx-select {
-      .SelectFacade(invalid-focus);
+      @include hxSelectFacade(invalid, $focused: true);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/index.scss]: Please change to:

```scss
@include hxSelectFacade(invalid, true);
```

File: src/scss/components/dropdown-select/index.scss [hunk 33]
```diff
@@ -70,74 +65,73 @@ hx-select-control {
...
+  @include hxSelectControl(invalid);
 
   ~ hx-select {
-    .SelectFacade(invalid);
+    @include hxSelectFacade(invalid);
   }
 
   &:focus {
-    .SelectControl(invalid-focus);
+    @include hxSelectControl(invalid, $focused: true);
```
Comment (100stacks) [inline src/scss/components/dropdown-select/index.scss]: Please change to:

```scss
@include hxSelectControl(invalid, true);
```

File: src/scss/components/dropdown-select/_config.scss [hunk 34]
```diff
@@ -4,6 +4,10 @@
 
 /// ========== MIXINS ========== ///
 @import "mixins";
```
Comment (100stacks) [inline src/scss/components/dropdown-select/_config.scss]: It doesn't look like we're using.  Please remove.

**Pull request merged**